File: python/strategy/Hunter_game.py


class HunterGame:

    def __init__(self, gun, tool, target) -> None:
        self.gun = gun
        self.tool = tool
        self.target = target

    def play(self):
        print("""
        ###########   Game Starts   ###########
        
        """)
        job_done = False
        self.tool.use(self.target)
        self.gun.load()
        for i in range(10):
            self.gun.shoot(self.target)
            if not self.target.is_alive():
                job_done = True
                break

        if not job_done:
            print("Unlucky day :(")
        else:
            print("Dinner will be served!")

File: c++/strategy/src/Hunter_game.cpp
#include "Hunter_game.h"

using namespace std;

void HunterGame::play() {
    cout << 
    "\n " <<
    "###########   Game Starts   ########### \n" <<
    "\n" << endl;

    bool job_done = false;
    _tool->use(_target);
    _gun->load();
    for (int i = 0; i < 10; i++) {
        _gun->shoot(_target);
        if (!_target->is_alive()) {
            job_done = true;
            break;
        }
    }

    if (!job_done)
        cout << "Unlucky day :(" << endl;
    else
        cout << "Dinner will be served!" << endl;
}